From 15b8bf99983e9b4b4248f70b63dc4ba28c92b75c Mon Sep 17 00:00:00 2001
From: AppaRao Puli <apparao.puli@intel.com>
Date: Tue, 18 Jul 2023 17:46:31 +0000
Subject: [PATCH] Add CTL build flag and disable BiosConfig

Some of the configs are not applicable to CTL. So Added
build flag and disabled Bios Config node.

Signed-off-by: AppaRao Puli <apparao.puli@intel.com>
---
 meson.build                      | 1 +
 meson_options.txt                | 7 +++++++
 redfish-core/include/redfish.hpp | 2 ++
 redfish-core/lib/systems.hpp     | 2 ++
 4 files changed, 12 insertions(+)

diff --git a/meson.build b/meson.build
index c5e27724..0ebfc1d1 100644
--- a/meson.build
+++ b/meson.build
@@ -96,6 +96,7 @@ feature_map = {
   'vm-websocket'                                : '-DBMCWEB_ENABLE_VM_WEBSOCKET',
   'xtoken-auth'                                 : '-DBMCWEB_ENABLE_XTOKEN_AUTHENTICATION',
   'vm-nbdproxy'                                 : '-DBMCWEB_ENABLE_VM_NBDPROXY',
+  'redfish-ctl-disable-routes'                  : '-DBMCWEB_CTL_DISABLE_ROUTES',
 }
 
 # Get the options status and build a project summary to show which flags are
diff --git a/meson_options.txt b/meson_options.txt
index baeff269..bea5f6b5 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -380,3 +380,10 @@ option(
                     enable on production systems at this time.  Other query
                     parameters such as only are not controlled by this option.'''
 )
+option(
+    'redfish-ctl-disable-routes',
+    type: 'feature',
+    value: 'enabled',
+    description: '''Disable Redfish service request routes.
+                    as these service are disabled for ctl-platfrom.'''
+)
diff --git a/redfish-core/include/redfish.hpp b/redfish-core/include/redfish.hpp
index 247cc0fe..f66b9d6d 100644
--- a/redfish-core/include/redfish.hpp
+++ b/redfish-core/include/redfish.hpp
@@ -225,11 +225,13 @@ class RedfishService
 
         requestRoutesSystems(app);
 
+#ifndef BMCWEB_CTL_DISABLE_ROUTES
         requestRoutesBiosService(app);
         requestRoutesBiosReset(app);
         requestRoutesBiosSettings(app);
         requestRoutesBiosAttributeRegistry(app);
         requestRoutesBiosChangePassword(app);
+#endif //BMCWEB_CTL_DISABLE_ROUTES
 
 #ifdef BMCWEB_ENABLE_VM_NBDPROXY
         requestNBDVirtualMediaRoutes(app);
diff --git a/redfish-core/lib/systems.hpp b/redfish-core/lib/systems.hpp
index 5b860e06..3d9a9b94 100644
--- a/redfish-core/lib/systems.hpp
+++ b/redfish-core/lib/systems.hpp
@@ -3402,8 +3402,10 @@ inline void
 
     asyncResp->res.jsonValue["LogServices"]["@odata.id"] =
         "/redfish/v1/Systems/system/LogServices";
+#ifndef BMCWEB_CTL_DISABLE_ROUTES
     asyncResp->res.jsonValue["Bios"]["@odata.id"] =
         "/redfish/v1/Systems/system/Bios";
+#endif
 
     asyncResp->res.jsonValue["Oem"]["VoltageRegulators"] = {
         {"@odata.id", "/redfish/v1/Systems/system/Oem/VoltageRegulators"}};
-- 
2.25.1

